x/auth/types/account_retriever.go

```diff
@@ -0,0 +1,65 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NodeQuerier is an interface that is satisfied by types that provide the QueryWithData method
+type NodeQuerier interface {
+	// QueryWithData performs a query to a Tendermint node with the provided path
+	// and a data payload. It returns the result and height of the query upon success
+	// or an error if the query fails.
+	QueryWithData(path string, data []byte) ([]byte, int64, error)
+}
+
+// AccountRetriever defines the properties of a type that can be used to
+// retrieve accounts.
+type AccountRetriever struct {
```

Comment:
why do we need an extra wrapper for NodeQuerier?

Reply:
`NodeQuerier` is an interface that could be satisfied by any type providing a `QueryWithData` functionality. `AccountRetriever` is the `auth`-specific implementation to be used in current clients.

Reply:
It also removes the dependency on the `client/` package.